src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/SharedBlobListenerFactory.cs

```diff
@@ -13,8 +13,9 @@ internal class SharedBlobListenerFactory : IFactory<SharedBlobListener>
         private readonly IStorageAccount _account;
         private readonly IWebJobsExceptionHandler _exceptionHandler;
         private readonly IContextSetter<IBlobWrittenWatcher> _blobWrittenWatcherSetter;
+        private readonly string _hostId;
 
-        public SharedBlobListenerFactory(IStorageAccount account,
+        public SharedBlobListenerFactory(string hostId, IStorageAccount account,
```

Comment:
To clarify a bit more - only one instance does the log scanning and population of the blob trigger queue. However ALL instances listen to that queue, so blob processing itself is parallelized. I did that because we don't want this relatively expensive scanning to be duplicated on all instances.